games-misc/moonlight-embedded/moonlight_embedded-20240611~git.recipe

```diff
@@ -0,0 +1,95 @@
+SUMMARY="Moonlight Embedded Gamestream client for embedded systems"
+DESCRIPTION="Moonlight Embedded is an open source client for Sunshine and NVIDIA GameStream for \
+embedded Linux systems, like Raspberry Pi, CuBox-i and ODROID. Moonlight allows you to stream your \ 
+full collection of games and applications from your PC to other devices to play them remotely."
+HOMEPAGE="https://github.com/moonlight-stream/moonlight-embedded"
+COPYRIGHT="2015-2024 Iwan Timmer"
+LICENSE="GNU GPL v3"
+REVISION="1"
+# Main moonlight-embedded
+srcGitRev="2782ec6a79ac0cecf316ad361de3f28a1fa80393"
+SOURCE_URI="https://github.com/robante15/moonlight-embedded/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="b57d1ee4ba15adf3ad217f03a4fa0bc105f787aab0a84ca2cca3bd690e2ee31c"
+SOURCE_DIR="moonlight-embedded-$srcGitRev"
+
+# moonlight-common-c
+srcGitRev_2="8599b6042a4ba27749b0f94134dd614b4328a9bc"
+SOURCE_URI_2="https://github.com/moonlight-stream/moonlight-common-c/archive/$srcGitRev_2.tar.gz"
+CHECKSUM_SHA256_2="1d62bdcd81e77b8bf6dc804c7402685e806dbcc999240857bbaf62790b24fdbc"
+
+#enet
+srcGitRev_3="bbf71856bb144729af4ed08af6bc2f5826a96db5"
+SOURCE_URI_3="https://github.com/cgutman/enet/archive/$srcGitRev_3.tar.gz"
+CHECKSUM_SHA256_3="b55531ba56bfad5bb7bef606d4ea4868cc50e1e38cc4778bdec23edd4ee5c2ab"
+
+#enet
+srcGitRev_4="ae51c9942f0d985b2161d6b22986f041fc98ce5c"
+SOURCE_URI_4="https://github.com/mdqinc/SDL_GameControllerDB/archive/$srcGitRev_4.tar.gz"
+CHECKSUM_SHA256_4="fd279f3a60434d139564dec7f77bb8af81ccd129bce9aa474234762845f8c601"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	moonlight_embedded$secondaryArchSuffix = $portVersion
+	cmd:moonlight = $portVersion
+	"
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libopus$secondaryArchSuffix
+	lib:libavcodec$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libexpat$secondaryArchSuffix
+	lib:libavutil$secondaryArchSuffix
+	lib:libuuid$secondaryArchSuffix
+	lib:libSDL2_2.0$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	ffmpeg5${secondaryArchSuffix}_devel
```

Comment:
maybe replace this with `devel:libavcodec$secondaryArchSuffix >= 59` and `devel:libavutil$secondaryArchSuffix >= 57`?

Also, these and the REQUIRES above should be sorted alphabetically. The `haiku` entry can stay first.

Reply:
As mentioned by @jmairboeck 

The problem probably is that you have another ffmpeg devel package installed and it conflicts with ffmpeg5_devel, haikuporter should give a notice on which one that is, uninstall this on your system and the build should proceed.

Reply:
When I stopped requiring ffmpeg5 and used the requirements of `devel:libavcodec$secondaryArchSuffix >= 59` and `devel:libavutil$secondaryArchSuffix >= 5` I already managed to ensure that haikuports did not throw any errors.

Now the problem is when I try to compile with gcc2 I get compilation errors. Architectures that I use in my recipe:
```
ARCHITECTURES="all"
SECONDARY_ARCHITECTURES="x86_gcc2 x86"
```

Reply:
You can keep gcc2 disabled.

```
ARCHITECTURES="all !x86_gcc2"
SECONDARY_ARCHITECTURES="x86"
```

Reply:
Okay if I leave the architectures like that I get this error

```
~> haikuporter -S moonlight_embedded-20240611~git
Checking if any dependency-infos need to be updated ...
Looking for stale dependency-infos ...
Warning: Port moonlight_embedded-20240611~git is broken on this architecture.
Continue (y/n + enter)? y
 ok
Warning: building obsolete port, packages will be put in /boot/home/Desktop/Repos/haikuports/packages/.obsolete
Checking if any dependency-infos need to be updated ...
Looking for stale dependency-infos ...
----------------------------------------------------------------------
games-misc::moonlight_embedded-20240611~git
        /boot/home/Desktop/Repos/haikuports/games-misc/moonlight_embedded/moonlight_embedded-20240611~git.recipe
----------------------------------------------------------------------
Skipping download of source for 2782ec6a79ac0cecf316ad361de3f28a1fa80393.tar.gz
Skipping checksum validation of 2782ec6a79ac0cecf316ad361de3f28a1fa80393.tar.gz
Skipping download of source for 8599b6042a4ba27749b0f94134dd614b4328a9bc.tar.gz
Skipping checksum validation of 8599b6042a4ba27749b0f94134dd614b4328a9bc.tar.gz
Skipping download of source for bbf71856bb144729af4ed08af6bc2f5826a96db5.tar.gz
Skipping checksum validation of bbf71856bb144729af4ed08af6bc2f5826a96db5.tar.gz
Skipping download of source for ae51c9942f0d985b2161d6b22986f041fc98ce5c.tar.gz
Skipping checksum validation of ae51c9942f0d985b2161d6b22986f041fc98ce5c.tar.gz
Skipping unpack of 2782ec6a79ac0cecf316ad361de3f28a1fa80393.tar.gz
Skipping unpack of 8599b6042a4ba27749b0f94134dd614b4328a9bc.tar.gz
Skipping unpack of bbf71856bb144729af4ed08af6bc2f5826a96db5.tar.gz
Skipping unpack of ae51c9942f0d985b2161d6b22986f041fc98ce5c.tar.gz
Cleaning up temporary directories ...
'NoneType' object is not iterable
```

Reply:
`hp moonlight_embeded_x86` should do the trick (don't have 32bit running atm)

Reply:
I'm sorry if I didn't understand the comment, but I get the following

```
~> haikuporter -S moonlight_embedded_x86
Checking if any dependency-infos need to be updated ...
Looking for stale dependency-infos ...
Error: moonlight_embedded_x86 not found in repository
~> haikuporter -S moonlight_embedded_x86
Checking if any dependency-infos need to be updated ...
Looking for stale dependency-infos ...
Error: moonlight_embedded_x86 not found in repository
~> hp moonlight_embeded_x86
bash: hp: orden no encontrada
```

Reply:
Maybe you need to enable this in your config file:

https://github.com/haikuports/haikuporter/blob/master/haikuports-sample.conf#L73

Reply:
Thank you very much, I already did it, but in that case I once again have the problems with the libraries coming from ffmpeg (devel:libavcodec$secondaryArchSuffix >= 59 and devel:libavutil$secondaryArchSuffix >= 57), and even if I try to require ffmpeg itself I get the same error

Error:  Reason: build-requires "devel:libavcodec_x86 >= 59" of package "moonlight_embedded_x86-20240611~git" could not be resolved

Reply:
Works here, could you post the full error? I'm still thinking about conflicting devel package.

Reply:
```
~> haikuporter moonlight_embedded_x86
Checking if any dependency-infos need to be updated ...
Looking for stale dependency-infos ...
----------------------------------------------------------------------
games-misc::moonlight_embedded_x86-20240611~git
        /boot/home/Desktop/Repos/haikuports/games-misc/moonlight_embedded/moonlight_embedded-20240611~git.recipe
----------------------------------------------------------------------
Skipping download of source for 2782ec6a79ac0cecf316ad361de3f28a1fa80393.tar.gz
Skipping checksum validation of 2782ec6a79ac0cecf316ad361de3f28a1fa80393.tar.gz
Skipping download of source for 8599b6042a4ba27749b0f94134dd614b4328a9bc.tar.gz
Skipping checksum validation of 8599b6042a4ba27749b0f94134dd614b4328a9bc.tar.gz
Skipping download of source for bbf71856bb144729af4ed08af6bc2f5826a96db5.tar.gz
Skipping checksum validation of bbf71856bb144729af4ed08af6bc2f5826a96db5.tar.gz
Skipping download of source for ae51c9942f0d985b2161d6b22986f041fc98ce5c.tar.gz
Skipping checksum validation of ae51c9942f0d985b2161d6b22986f041fc98ce5c.tar.gz
Skipping unpack of 2782ec6a79ac0cecf316ad361de3f28a1fa80393.tar.gz
Skipping unpack of 8599b6042a4ba27749b0f94134dd614b4328a9bc.tar.gz
Skipping unpack of bbf71856bb144729af4ed08af6bc2f5826a96db5.tar.gz
Skipping unpack of ae51c9942f0d985b2161d6b22986f041fc98ce5c.tar.gz
Cleaning up temporary directories ...
build-requires "devel:libavcodec_x86 >= 59" of package "moonlight_embedded_x86-20240611~git" could not be resolved
Error: unable to resolve required packages for build for moonlight_embedded_x86-20240611~git
Error:  dependency-infos:
Error:          /boot/home/Desktop/Repos/haikuports/repository/moonlight_embedded_x86-20240611~git.DependencyInfo
Error:  repositories:
Error:          ['/boot/home/Desktop/Repos/haikuports/packages', '/boot/system/packages']
Error:  Reason: build-requires "devel:libavcodec_x86 >= 59" of package "moonlight_embedded_x86-20240611~git" could not be resolved
```

Recipe:

```
ARCHITECTURES="all !x86_gcc2"
SECONDARY_ARCHITECTURES="x86"
```

```
BUILD_REQUIRES="
	haiku${secondaryArchSuffix}_devel
	#ffmpeg5${secondaryArchSuffix}_devel
	devel:libavcodec$secondaryArchSuffix >= 59
	devel:libavutil$secondaryArchSuffix >= 57
	devel:libcurl$secondaryArchSuffix
	devel:libexpat$secondaryArchSuffix
	devel:libopus$secondaryArchSuffix
	devel:libsdl2$secondaryArchSuffix
	devel:libuuid$secondaryArchSuffix
	"
```

Reply:
That's weird, I don't see the error why it should fail.
Could you try (in Terminal):

	ls /system/lib/x86/libavcodec*

	pkgman search ffmpeg

Reply:
```
~> ls /system/lib/x86/libavcodec*
/system/lib/x86/libavcodec.so.60  /system/lib/x86/libavcodec.so.60.31.102
~> pkgman search ffmpeg
Status  Name                          Description                               
--------------------------------------------------------------------------------
        ffmpeg                        Audio and video recording, conversion, and
        ffmpeg_avdevice               Audio and video recording, conversion, and
        ffmpeg_debuginfo              Audio and video recording, conversion, and
        ffmpeg_devel                  Audio and video recording, conversion, and
        ffmpeg_source                 Audio and video recording, conversion, and
        ffmpeg_tools                  Audio and video recording, conversion, and
        ffmpeg_x86                    Audio and video recording, conversion, and
        ffmpeg_x86_avdevice           Audio and video recording, conversion, and
        ffmpeg_x86_debuginfo          Audio and video recording, conversion, and
        ffmpeg_x86_devel              Audio and video recording, conversion, and
        ffmpeg_x86_source             Audio and video recording, conversion, and
        ffmpeg_x86_tools              Audio and video recording, conversion, and
S       ffmpeg5                       Audio and video recording, conversion, and
S       ffmpeg5_avdevice              Audio and video recording, conversion, and
        ffmpeg5_debuginfo             Audio and video recording, conversion, and
S       ffmpeg5_devel                 Audio and video recording, conversion, and
        ffmpeg5_source                Audio and video recording, conversion, and
        ffmpeg5_tools                 Audio and video recording, conversion, and
        ffmpeg5_x86                   Audio and video recording, conversion, and
        ffmpeg5_x86_avdevice          Audio and video recording, conversion, and
        ffmpeg5_x86_debuginfo         Audio and video recording, conversion, and
        ffmpeg5_x86_devel             Audio and video recording, conversion, and
        ffmpeg5_x86_source            Audio and video recording, conversion, and
        ffmpeg5_x86_tools             Audio and video recording, conversion, and
S       ffmpeg6                       Audio and video recording, conversion, and
        ffmpeg6_avdevice              Audio and video recording, conversion, and
        ffmpeg6_debuginfo             Audio and video recording, conversion, and
        ffmpeg6_devel                 Audio and video recording, conversion, and
        ffmpeg6_source                Audio and video recording, conversion, and
        ffmpeg6_tools                 Audio and video recording, conversion, and
S       ffmpeg6_x86                   Audio and video recording, conversion, and
        ffmpeg6_x86_avdevice          Audio and video recording, conversion, and
        ffmpeg6_x86_debuginfo         Audio and video recording, conversion, and
        ffmpeg6_x86_devel             Audio and video recording, conversion, and
        ffmpeg6_x86_source            Audio and video recording, conversion, and
        ffmpeg6_x86_tools             Audio and video recording, conversion, and
        ffmpeggui_x86                 A graphical front-end for ffmpeg          
        ffmpeggui_x86_source          A graphical front-end for ffmpeg (source f
        ffmpegthumbnailer_x86         A lightweight video thumbnailer           
        ffmpegthumbnailer_x86_devel   A lightweight video thumbnailer (developme
        ffmpegthumbnailer_x86_source  A lightweight video thumbnailer (source fi
        qwinff_x86                    A GUI for FFMPEG                          
        qwinff_x86_source             A GUI for FFMPEG (source files)           
        sdl_kitchensink_x86           FFmpeg and SDL2 based library for audio an
        sdl_kitchensink_x86_devel     FFmpeg and SDL2 based library for audio an
        sdl_kitchensink_x86_source    FFmpeg and SDL2 based library for audio an
~>
```

Reply:
My bad, this should be: `ls /system/develop/lib/x86/libavcodec*`
I'm guessing you installed "haikuporter" from the depot? eg not cloned haikuporter from it's repo?

Reply:
I already managed to compile it, I had the wrong packages installed, I corrected that and it compiled the x86 version without problems, I also did a test run and it connected to the server.

Reply:
Maybe just a last notice, change the ffmpeg5 dependencies as you mentioned in a reply and as suggested by @jmairboeck